mlir-sycl/lib/Conversion/SYCLToGPU/SYCLToGPU.cpp

```diff
@@ -0,0 +1,223 @@
+//===- SYCLToGPU.cpp - SYCL to GPU Patterns -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements patterns to convert SYCL dialect to GPU dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/SYCLToGPU/SYCLToGPU.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/SYCL/IR/SYCLOps.h"
+#include "mlir/Support/LLVM.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace mlir::sycl;
+
+namespace {
+/// Returns the result of creating an operation to get a reference to an element
+/// of a sycl::id or sycl::range.
+Value createGetOp(OpBuilder &builder, Location loc, Type dimMtTy, Value res,
```

Comment:
What does `dimMtTy` stand for?

Reply:
Bad name, for sure. It's the type of the underlying array for the input type. Changing it now.